src/installer.js

```diff
@@ -1,32 +1,100 @@
 const {exec} = require('@actions/exec')
 const path = require('path')
-const semver = require('semver')
+const core = require('@actions/core')
 
-module.exports = {installElixir, installOTP}
+/**
+ * Install Erlang/OTP.
+ *
+ * @param {string} osVersion
+ * @param {string} otpVersion
+ */
+async function installOTP(osVersion, otpVersion) {
+  return new Promise(async (resolve, reject) => {
+    const maybePlatformIsOKError = checkPlatform()
+    if (maybePlatformIsOKError !== 'ok') {
+      reject(new Error(maybePlatformIsOKError))
+      return
+    }
+    await exec(path.join(__dirname, 'install-otp'), [osVersion, otpVersion])
+      .then(() => {
+        resolve()
+      })
+      .catch((err) => {
+        core.warning(
+          `Installing Erlang/OTP ${otpVersion} over ${osVersion} failed. ` +
+            'It is possible the requested version does not exist',
+        )
+        reject(err)
+      })
+  })
+}
 
 /**
  * Install Elixir.
  *
- * @param {string} version
- * @param {string} otpMajor
+ * @param {string} elixirVersion
  */
-async function installElixir(version, otpMajor) {
-  if (process.platform === 'linux') {
-    const otpString = otpMajor ? `-otp-${otpMajor}` : ''
-    await exec(path.join(__dirname, 'install-elixir'), [version, otpString])
-  }
+async function installElixir(elixirVersion) {
+  return new Promise(async (resolve, reject) => {
+    const maybePlatformIsOKError = checkPlatform()
+    if (maybePlatformIsOKError !== 'ok') {
+      reject(new Error(maybePlatformIsOKError))
+      return
+    }
+    await exec(path.join(__dirname, 'install-elixir'), [elixirVersion])
+      .then(() => {
+        resolve()
+      })
+      .catch((err) => {
+        core.warning(
+          `Installing Elixir ${elixirVersion} failed. ` +
+            'It is possible the requested version does not exist',
+        )
+        reject(err)
+      })
+  })
 }
 
 /**
- * Install OTP.
+ * Install rebar3.
  *
- * @param {string} version
+ * @param {string} rebar3Version
  */
-async function installOTP(version, osVersion) {
-  if (process.platform === 'linux') {
-    await exec(path.join(__dirname, 'install-otp'), [version, osVersion])
-    return
+async function installRebar3(rebar3Version) {
+  return new Promise(async (resolve, reject) => {
+    const maybePlatformIsOKError = checkPlatform()
+    if (maybePlatformIsOKError !== 'ok') {
+      reject(new Error(maybePlatformIsOKError))
+      return
+    }
+    await exec(path.join(__dirname, 'install-rebar3'), [rebar3Version])
+      .then(() => {
+        resolve()
+      })
+      .catch((err) => {
+        core.warning(
+          `Installing rebar3 ${rebar3Version} failed. ` +
+            'It is possible the requested version does not exist',
+        )
+        reject(err)
+      })
+  })
+}
+
+function checkPlatform() {
+  if (
+    process.platform !== 'linux' &&
+    process.env.CI_TEST_ENV !== 'test' &&
```

Comment:
What is this check doing?

Reply:
Making sure it's either Linux or Darwin in testing. I'll revert it. I don't want the PR to be blocked for this.

Reply:
This is reverted.

Reply:
I was asking specifically about `process.env.CI_TEST_ENV !== 'test'`. What does that check do?

Reply:
Nothing, anymore, since it was eventually removed. At a given point in time I had it defined as an env. variable.